File: 03 Podminky a cykly/07 csv.py
import json
import csv
import random
JSON_FILE = "_data.json"
CSV_FILE = "_soubor.csv"

with open(JSON_FILE, encoding="utf-8", newline='') as file:
    data = json.load(file)
    jmena = data['jmena']
    kapely = data['kapely']

with open(CSV_FILE, encoding="utf-8", newline='') as file:
    rows = list(csv.DictReader(file))

rows.append({
    'name': random.choice(jmena),
    'age': random.randint(0, 99),
    'favorite_band': random.choice(kapely)
})

with open(CSV_FILE, 'w', encoding="utf-8", newline='') as file:
    keys = list(rows[0].keys())
    writer = csv.DictWriter(file, fieldnames=keys)
    writer.writeheader()
    writer.writerows(rows)

File: 03 Podminky a cykly/07 csv.ts
import { serializeCsv, deserializeCsv } from "./07%20csv%20funkce.ts"
import jmenaAKapely from "./_data.json" assert { type: "json" };
const { jmena, kapely } = jmenaAKapely;
const filename = "_soubor.csv";

function vyberNahodnyElement(arr: string[]): string {
  const index = Math.floor(Math.random() * arr.length);
  return arr[index];
}

const text = await Deno.readTextFile(filename);
const tabulka = deserializeCsv(text);

tabulka.push({
  name: vyberNahodnyElement(jmena),
  age: Math.floor(Math.random() * 100).toString(),
  favorite_band: vyberNahodnyElement(kapely),
});

await Deno.writeTextFile(filename, serializeCsv(tabulka));
